src/components/record/record.tsx

```diff
@@ -28,36 +36,180 @@ const Record = ({
 };
 
 const RecordInner = () : JSX.Element => {
-  const { page, readMainEntity } = useRecord();
+  const { page, readMainEntity, reference } = useRecord();
+
+  /**
+   * State variable to show or hide side panel
+   */
+  const [ showPanel, setShowPanel ] = useState<boolean>(true);
+
+  const [containerHeight, setContainerHeight] = useState<number>(0);
+
+  /**
+   * Function calculates available height in the window to fit record container
+   */
+   const calculateAvailableHeight = () => {
+    const windowHeight = window.innerHeight;
+    const container = document.getElementsByClassName('main-container')?.[0];
+    if (container) {
+      const containerTop = container.getBoundingClientRect().y;
+
+      const availableHeight = (100 - (containerTop / windowHeight) * 100);
+
+      setContainerHeight(availableHeight);
+    }
+  }
 
   useEffect(() => {
     // get the data
     readMainEntity();
   }, []);
 
+  useLayoutEffect(() => {
+    calculateAvailableHeight();
```

Comment:
There's a `attachContainerHeightSensors` function in `ui-utils.ts` that is basically the React version of the same function that we have in AngularJS. So please use this function instead. You don't need to pass any parameters to this function for your usecase. Please take a look at how I did it in recordset page.